macro-plantuml-macro/src/main/java/org/xwiki/contrib/plantuml/internal/DefaultPlantUMLConfiguration.java

```diff
@@ -56,4 +56,16 @@ public String getPlantUMLServerURL()
         }
         return serverURL;
     }
+
+    @Override
+    public String getDefaultOutputFormat() {
+        String format = this.plantUMLConfigurationSource.getProperty("format");
+        // The returned value can be null if no xobject has been defined on the wiki config page.
+        if (format == null) {
+            // Fallback to xwiki.properties
+            format = this.xwikiPropertiesConfigurationSource.getProperty("plantuml.format", "png");
```

Comment:
Note: my previous feedback was not applied. You can write:

`return this.plantUMLConfigurationSource.getProperty("format", this.xwikiPropertiesConfigurationSource.getProperty("plantuml.format", "png");`